osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -223,16 +190,13 @@ private void cancelTrackedBindableUpdate()
         {
             lock (bindableUpdateLock)
             {
-                trackedUpdateCancellationSource?.Cancel();
-                trackedUpdateCancellationSource = null;
+                trackedUpdateCancellationSource.Cancel();
+                trackedUpdateCancellationSource = new CancellationTokenSource();
 
-                if (linkedCancellationSources != null)
```

Comment:
Commentary: This null check blames back to 46483622149904c9240f6f6e53ba4ef283edb07d and was related to finalisation woes. But this *should* no longer be a concern since https://github.com/ppy/osu-framework/pull/4242. Same goes for the other change in `Dispose()` below.